pkg/scheduler/objects/application.go

```diff
@@ -2009,3 +2018,20 @@ func (sa *Application) SetTimedOutPlaceholder(taskGroupName string, timedOut int
 		sa.placeholderData[taskGroupName].TimedOut = timedOut
 	}
 }
+
+func (sa *Application) sendRemoveAskEvent(allocKey string) {
+	sa.RLock()
+	defer sa.RUnlock()
```

Comment:
This lock must be eliminated because the method is called from a locked call site. If it's needed then you have to create another one without locks.

Reply:
Removed it. Thanks!